nodestream/interpreting/interpretations/switch_interpretation.py

```diff
@@ -28,67 +40,79 @@ class SwitchInterpretation(Interpretation, alias="switch"):
     )
 
     @staticmethod
-    def guarantee_interpretation_list_from_file_data(file_data) -> List[Interpretation]:
+    def guarantee_single_interpretation_pass_from_file_data(
+        file_data,
+    ) -> InterpretationPass:
+        # Check for multiinterpretationpass.
         if isinstance(file_data, list):
-            return [
-                Interpretation.from_file_data(**interpretation)
-                for interpretation in file_data
-            ]
+            interpretation_pass = InterpretationPass.from_file_data(file_data)
+        else:
+            interpretation_pass = InterpretationPass.from_file_data([file_data])
+
+        # We do not support multiple interpretation passes within the switch interpretation.
+        if isinstance(interpretation_pass, MultiSequenceInterpretationPass):
+            raise SwitchError(INVALID_SWITCH_ERROR_MESSAGE)
 
-        return [Interpretation.from_file_data(**file_data)]
+        return interpretation_pass
 
     def __init__(
         self,
         switch_on: StaticValueOrValueProvider,
-        cases: Dict[str, dict],
-        default: Dict[str, Any] = None,
+        cases: Dict[str, Interpretation | List[Interpretation]],
+        default: Interpretation | List[Interpretation] = None,
         normalization: Dict[str, Any] = None,
         fail_on_unhandled: bool = True,
     ):
         self.switch_on = ValueProvider.guarantee_value_provider(switch_on)
-        self.interpretations = {
-            field_value: self.guarantee_interpretation_list_from_file_data(
+        self.branches = {
+            field_value: self.guarantee_single_interpretation_pass_from_file_data(
                 interpretation
             )
             for field_value, interpretation in cases.items()
         }
         self.default = (
-            self.guarantee_interpretation_list_from_file_data(default)
+            self.guarantee_single_interpretation_pass_from_file_data(default)
             if default
             else None
         )
         self.normalization = normalization or {}
         self.fail_on_unhandled = fail_on_unhandled
+        self.verify_completeness()
 
     @property
-    def distinct_branches(self) -> bool:
+    def all_interpretations(self) -> Iterable[Interpretation]:
+        yield from list(self.branches.values()) + (
+            [self.default] if self.default is not None else []
+        )
```

Comment:
This appears clearer to me what is happening. 

```suggestion
        yield from self.branches.values()
        if self.default:
            yield self.default
```